src/components/data-pane/wildcard-field-drop-zone.tsx

```diff
@@ -0,0 +1,90 @@
+import {Schema} from 'compassql/build/src/schema';
+import {isWildcard} from 'compassql/build/src/wildcard';
+import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
+import {ConnectDropTarget, DropTarget, DropTargetCollector, DropTargetSpec} from 'react-dnd';
+import {CUSTOM_WILDCARD_ADD, CustomWildcardAction} from '../../actions/custom-wildcard-field';
+import {DraggableType} from '../../constants';
+import {DraggedFieldIdentifier} from '../field/index';
+import * as styles from './wildcard-field-drop-zone.scss';
+
+
+export interface CustomWildcardFieldDropZoneDropTargetProps {
+  connectDropTarget: ConnectDropTarget;
+  isOver: boolean;
+  item: Object;
+  canDrop: boolean;
+}
+
+export interface CustomWildcardFieldDropZonePropsBase {
+  schema: Schema;
+  handleAction?: (action: CustomWildcardAction) => void;
+}
+
+interface CustomWildcardFieldDropZoneProps extends
+  CustomWildcardFieldDropZoneDropTargetProps, CustomWildcardFieldDropZonePropsBase {};
+
+class CustomWildcardFieldDropZoneBase extends React.PureComponent<CustomWildcardFieldDropZoneProps, {}> {
+  public constructor(props: CustomWildcardFieldDropZoneProps) {
+    super(props);
+  }
+
+  public render() {
+    const {connectDropTarget} = this.props;
+
+    return connectDropTarget(
+      <div className='drop-zone'>
+        Drop a field here!
+      </div>
+    );
+  }
+}
+
+
+const customWildcardFieldTarget: DropTargetSpec<CustomWildcardFieldDropZoneProps> = {
+  drop(props, monitor) {
+    if (monitor.didDrop()) {
+      return;
+    }
+
+    const {fieldDef} = monitor.getItem() as DraggedFieldIdentifier;
```

Comment:
Why do we need `as` here?

Reply:
Bad type inference. Monitor.getItem() shows up as type 'Object' even though that object contains fieldDef property. I casted to DraggedFieldIdentifier to get fieldDef. (We also the same thing in `encoding-shelf.tsx` for the `encodingShelfTarget`.